### PR TITLE
Update Columns Being Displayed

### PR DESCRIPTION
**Pull request by kpattaswamy**
This PR presets all column options as chosen to display when a script is selected. Removing a particular column that is being displayed will safely remove that column's usage in a potential groupby/aggregate query as well. 

Also renamed the column_filtering.tsx file to column_display.tsx for better representation and moved corresponding functionality to that file from query_editor.tsx.

### CODE REVIEW

File: src/column_display.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+): string {
+  // Setting script default to all the columns in the script
+  let script = allColumnOptions.map(({ label }) => `'${label}'`).join();
+
+  // Updating the script if the user selected columns to filter
+  if (chosenOptions?.length > 0) {
+    script = chosenOptions.map(({ label }) => `'${label}'`).join();
+  }
+
+  return script;
```
Comment (NickLanam) [inline src/column_display.tsx:40]:
```suggestion
  // If the user hasn't filtered the options yet, show all of them.
  const source = chosenOptions?.length ? chosenOptions : allColumnOptions;
  return source.map(({ label }) => `'${label}'`).join(',');
```

File: src/column_display.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+import React, { PureComponent } from 'react';
+import { MultiSelect, InlineLabel } from '@grafana/ui';
+import { QueryEditorProps, SelectableValue } from '@grafana/data';
+import { defaultQuery, PixieDataSourceOptions, PixieDataQuery } from './types';
+import { DataSource } from './datasource';
+
+type Props = QueryEditorProps<DataSource, PixieDataQuery, PixieDataSourceOptions>;
+
+export function getColumnsScript(
+  chosenOptions: Array<SelectableValue<number>>,
```
Comment (NickLanam) [inline src/column_display.tsx:29]: Nit for a future diff: `Array<SelectableValue<number>>` shows up so often in this plugin now that maybe we should alias it in `types.ts`, like `export type SelectableColumns = Array<SelectableValue<number>>` or some other useful name.

File: src/column_display.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    // Update Groupby Array with correct columns to display
+    const groupByArr = query.queryMeta?.selectedColGroupby?.filter(({ label }) => {
+      return label !== colToRemove;
+    });
+
+    // Update Aggregate Array with correct columns to display
+    const aggArr = query.queryMeta?.aggData?.filter(({ aggColumn }) => {
+      return aggColumn !== colToRemove;
+    });
+    const aggData = groupByArr?.length ? aggArr ?? [] : [];
```
Comment (NickLanam) [inline src/column_display.tsx:64]: Nit/clarity:

You can simplify the duplicate checks with a [Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set). Something like this:

```ts
const allowed = new Set(chosenOptions.map(({ label }) => label));

const groupByArr = query.queryMeta?.selectedColGroupby?.filter(
  ({ label}) => allowed.has(label)
) ?? [];
const groupLabels = new Set(groupByArr.map(({ label }) => label));

const aggData = query.queryMeta?.aggData?.filter(
  ({ aggColumn }) => groupLabels.has(aggColumn)
) ?? [];
```

File: src/column_display.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+    onChange({
+      ...query,
+      queryMeta: {
+        ...query.queryMeta,
+        selectedColDisplay: chosenOptions,
+        selectedColGroupby: groupByArr,
+        aggData: aggData,
```
Comment (NickLanam) [inline src/column_display.tsx:72]: Nit: shorthand syntax

```suggestion
        aggData,
```

**Review by NickLanam — APPROVED** [review]
LGTM with a few nits

**Review by aimichelle — APPROVED** [review]

**Pull request merged**